From baf12fe4e35d22fd1f7271b62c1e4304653c66f2 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Tue, 6 Nov 2018 15:30:40 +0100
Subject: [PATCH 3/4] [SQUSH][DNM] Merge tag 'android-9.0.0_r16' into
 lineage-16.0

commit 1ea2b2bb423ee2b16e5f02117ead086f3e00a193
Merge: b8d86c00 fa9ce0e4
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Aug 8 03:09:11 2018 +0000

    Snap for 4939496 from fa9ce0e4a240d50fde168369c2095d1e14241a97 to pi-qpr1-release

    Change-Id: I07a2fb093806a835d82f109574cc774d4a487da5

commit fa9ce0e4a240d50fde168369c2095d1e14241a97
Author: Greg Kaiser <gkaiser@google.com>
Date:   Wed Aug 1 12:47:48 2018 -0700

    Revert "Support Speck encryption."

    This is a partial revert of
    commit 6a0fa62dc71157f85b24ed15e39ada277f9b5201.

    Remove the Speck encryption support. It was eventually
    decided not to allow Speck in Android P, so this code
    is no longer needed and wasn't used outside of testing.

    We retain the e4crypt_get_policy_flags() changes from original
    commit, though.

    Bug: 112009351
    Test: Confirmed AES continues to work with FBE.
    Change-Id: I53b091dd4c5392de7a00a8c168919fb55e527eaf

commit b8d86c0051a0113659edf2bf7ea80b6d318e01f5
Author: Yabin Cui <yabinc@google.com>
Date:   Tue May 1 17:31:36 2018 -0700

    simpleperf: fix test on APCT.

    Bug: http://b/77470883
    Test: run adb shell
    Test: /data/local/tmp/simpleperf_unit_test/simpleperf_unit_test.
    Change-Id: I640ad76b68842045f6ecdcb3cf686b7823d7d5db
    (cherry picked from commit 03a31c13168f6502883e8beb4dbc76bc064a54a0)

Change-Id: I72bf28b75695875d1317a129d7f5e7fd4480c3b3
---
 simpleperf/cmd_record_test.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/simpleperf/cmd_record_test.cpp b/simpleperf/cmd_record_test.cpp
index 7f208862..31f17278 100644
--- a/simpleperf/cmd_record_test.cpp
+++ b/simpleperf/cmd_record_test.cpp
@@ -270,7 +270,10 @@ TEST(record_cmd, existing_threads) {
   ASSERT_TRUE(RunRecordCmd({"-t", tid_list}));
 }
 
-TEST(record_cmd, no_monitored_threads) { ASSERT_FALSE(RecordCmd()->Run({""})); }
+TEST(record_cmd, no_monitored_threads) {
+  TemporaryFile tmpfile;
+  ASSERT_FALSE(RecordCmd()->Run({"-o", tmpfile.path}));
+}
 
 TEST(record_cmd, more_than_one_event_types) {
   ASSERT_TRUE(RunRecordCmd({"-e", "cpu-cycles,cpu-clock"}));
-- 
2.17.1

